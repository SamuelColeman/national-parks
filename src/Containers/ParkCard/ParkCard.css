.park_card_text {
	padding: 0 15vh;
	color: #3C321F;
}

.park_card_title {
	font-size: 8vh;
	margin: 4vh;
	color: #585F29;
}

.park_card_p {
	font-size: 5vh;
	color: #585F29;
}

.park_card {
	text-decoration: none;
	border: 3px solid #3C321F;
  margin: 3vh;
  width: 50vh;
  height: 25vh;
  background: #AD9A7A;
  border-radius: 5px;
}

.park_card_t {
	text-decoration: none;
	color: #3C321F;
	font-size: 5vh;
  margin-bottom: 2%;
  margin-left: 2%;
  margin-right: 2%;
}

.park_card_d {
	text-decoration: none;
	text-underline-position: none;
	color: #3C321F;
	font-size: 3vh;
}

.park_section {
	display: grid;
  background: #AD9A7A;
  height: 100vh;
  text-align: center;
  width: 100%;
  grid-template-areas: 
		"title title title" 
		"form info info"
		"form info info";
	grid-template-columns: 25% auto;
	grid-template-rows: 20% auto;
}

.park_card_info {
	max-height: 100vh;
	overflow: scroll;
	grid-area: info;
}

.park_card_btns {
	grid-area: form;
	padding-top: 2vh;
	background-color: #585F29;
}

.park_card_header {
	grid-area: title;
	background-color: #3C321F;
}

.info_name {
	background: #3C321F;
	border: 1px solid #3C321F;
	width: 75%;
  margin: 4%;
  height: 5vh;
  font-size: 3vh;
  border-radius: 5px;
  color: #AD9A7A;
}

.info_name:hover {
	background: #AD9A7A;
	border: 1px solid #AD9A7A;
	color: #3C321F;
}