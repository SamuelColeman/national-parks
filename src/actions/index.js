export const hasError = (errorMsg) => ({
  type: 'HAS_ERROR',
  errorMsg
})

export const getParks = (parks) => ({
	type: 'GET_PARKS',
	parks
})

export const selectState = (selectedState) => ({
	type: 'SELECT_STATE',
	selectedState
})


export const getVisitorCenters = (visitorCenters) => ({
	type: 'GET_VISITOR_CENTERS',
	visitorCenters
})