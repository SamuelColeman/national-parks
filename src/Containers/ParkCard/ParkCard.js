import React from 'react';
import './ParkCard.css';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const ParkCard = ({ name, designation, id, parks, fullName, description, weatherInfo, page, parkCode, displayParkInfo, parkInfo, infoName }) => {
	if (page === false) {
		return (
			<Link to={`/parks/${id}`}>
				<section className='park_card'>
					<h2>{name}</h2>
					<h3>{designation}</h3>
				</section>
			</Link>
		)
	} else {
		return (
			<section>
				<h1 className='park_card_title'>{fullName}</h1>
				<h1 className='park_card_p'>Overview:</h1>
				<h2 className='park_card_text'>{description}</h2>
				<h1 className='park_card_p'>Weather:</h1>
				<h2 className='park_card_text'>{weatherInfo}</h2>
				<section onClick={(e) => displayParkInfo(parkCode, e)}>
					<Link to={`/parks/${id}/VisitorCenters`}>
						<button name='VisitorCenters'>Visitor Centers</button>
					</Link>
					<Link to={`/parks/${id}/Events`}>
						<button name='Events'>Events</button>
					</Link>
					<Link to={`/parks/${id}/Alerts`}>
						<button name='Alerts'>Alerts</button>
					</Link>
					<Link to={`/parks/${id}/Campgrounds`}>
						<button name='Campgrounds'>Campgrounds</button>
					</Link>
				</section>
			</section>
		)
	}
}

export const mapStateToProps = (state) => ({
  parks: state.parks,
  parkInfo: state.parkInfo,
  infoName: state.infoName
})

export default connect(mapStateToProps)(ParkCard);