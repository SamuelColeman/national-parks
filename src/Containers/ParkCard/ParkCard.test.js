import React from 'react';
import { shallow } from 'enzyme';
import { ParkCard, mapStateToProps, mapDispatchToProps } from './ParkCard';
import { getParkId } from '../../actions';

describe('ParkCard', () => {
	
	let wrapper;
	let mockPark = { 
    	description: "Historic Site",
			designation: "National Historic Site",
			directionsInfo: "Drive Car",
			directionsUrl: "http://www.directions.com",
			fullName: "National Historic Site",
			id: "2E",
			latLong: "lat:36, long:82",
			name: "Historic Park",
			parkCode: "pant",
			states: "ZN",
			url: "https://www.example.com",
			weatherInfo: "Sunny"
		};
	let mockParkInfo = [{
    	description: "Biggest visitor center ever.",
			directionsInfo: "Take a left.",
			directionsUrl: "https://www.exampledirections.com",
			id: "11",
			latLong: "{lat:35, lng:83}",
			name: "Visitor Center",
			parkCode: "pant",
			url: "https://www.examplecenter.com"
	}];
	let mockInfoName = 'VisitorCenters'
	let mockParks = [mockPark]
	let mockPage = false;
	let mockDisplayParkInfo = jest.fn();

	beforeEach(() => {
    wrapper = shallow(<ParkCard
    	{...mockPark} 
    	page={mockPage}
    	displayParkInfo={mockDisplayParkInfo}
    	/>)
 	})

  it('should match snapshot with correct data passing through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with correct data passing through', () => {
  	mockPage = true;
    expect(wrapper).toMatchSnapshot();
  });

  it('map state to props gives the parks array in state', () => {
		const mockState = { parks: mockParks};

		const expected = {
			parks: mockState.parks
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});

	it('map state to props gives the park info array in state', () => {
		const mockState = { parkInfo: mockParkInfo};

		const expected = {
			parkInfo: mockState.parkInfo
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});

	it('map state to props gives the info name in state', () => {
		const mockState = { infoName: mockInfoName};

		const expected = {
			infoName: mockState.infoName
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});

	it('map state to props gives the park id in state', () => {
		const mockState = { parkId: mockPark.id};

		const expected = {
			parkId: mockState.parkId
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});

	// it('calls dispatch with getParkId action when  is called', () => {
	// 		mockPage = false;
 //      const mockDispatch = jest.fn();
 //      const actionToDispatch = getParkId(mockPark.id);
 //      const mappedProps = mapDispatchToProps(mockDispatch);
 //      wrapper.find('Link').at(0).simulate('click');
 //      // mappedProps.handleSubmit('Learn Redux!', 1);

 //      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
 //    });
});