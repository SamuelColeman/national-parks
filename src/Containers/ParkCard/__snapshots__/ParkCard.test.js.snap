// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParkCard should match snapshot with correct data passing through 1`] = `
<Link
  onClick={[Function]}
  to="/parks/2E"
>
  <section
    className="park_card"
  >
    <h2
      className="park_card_t"
    >
      Historic Park
    </h2>
    <h3
      className="park_card_d"
    >
      National Historic Site
    </h3>
  </section>
</Link>
`;

exports[`ParkCard should match snapshot with correct data passing through 2`] = `
<Link
  onClick={[Function]}
  to="/parks/2E"
>
  <section
    className="park_card"
  >
    <h2
      className="park_card_t"
    >
      Historic Park
    </h2>
    <h3
      className="park_card_d"
    >
      National Historic Site
    </h3>
  </section>
</Link>
`;
